colossalai/tensor/param_op_hook.py

```diff
@@ -23,25 +28,78 @@ def post_backward(self, params: List[torch.Tensor]) -> None:
         pass
 
 
-class _ParamOpHookWrapper:
+class ParamOpHookManager:
+    """Manage your param op hooks. It only has static methods.
+    The only static method you should call is ``use_hooks(*hooks)``.
+    """
     hooks: Tuple[ParamOpHook, ...] = tuple()
 
+    @staticmethod
+    @contextmanager
+    def use_hooks(*hooks: ParamOpHook):
+        """Change the param op hooks you use. Nested calling is allowed.
+
+        Example::
+            >>> with ParamOpHookManager.use_hooks(*hooks):
+            >>>     do_something()
+            >>>     with ParamOpHookManager.use_hooks():
```

Comment:
why nest ParamOpHookManager context?

Reply:
In case some users want to nest. In most cases, we don't need to nest.

Reply:
I think users should not know what is OpHook. It is not a common-sense concept. Why do they need to have a manager?

Reply:
They can extend profiler by this manager. E.g. before op sample sth, after op sample sth. Indeed, it's for advanced developers.